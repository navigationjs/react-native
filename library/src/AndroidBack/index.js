import { Component } from 'react';
import navigation, { Navigation } from '../Navigation';
import Emitter from '../Navigation/Emitter';

export default class AndroidBack extends Component {
  componentDidMount() {
    const { id, onPress } = this.props;
    navigation.on(
      `${Navigation.EVENTS.ANDROID_BACK}${Emitter.SEPARATOR}${id}`,
      onPress
    );
  }

  componentWillUnmount() {
    const { id } = this.props;
    navigation.off(
      `${Navigation.EVENTS.ANDROID_BACK}${Emitter.SEPARATOR}${id}`
    );
  }

  render() {
    return null;
  }
}
