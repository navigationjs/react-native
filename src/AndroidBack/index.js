import { Component } from 'react';
import { BackHandler } from 'react-native';
import Back from '../Back';

export default class AndroidBack extends Component {
  componentDidMount() {
    this.backHandler = BackHandler.addEventListener(
      'hardwareBackPress',
      this.handleBackPress
    );
  }

  componentWillUnmount() {
    this.backHandler.remove();
  }

  handleBackPress = () => Back.back();

  render() {
    return null;
  }
}
