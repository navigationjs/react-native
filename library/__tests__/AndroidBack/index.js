import React from 'react';
import { render } from '@testing-library/react-native';
import AndroidBack from '../../src/AndroidBack';

describe('AndroidBack', () => {
  it('should match snapshot', async () => {
    const { baseElement } = render(
      <AndroidBack id={'example'} onPress={() => {}} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
