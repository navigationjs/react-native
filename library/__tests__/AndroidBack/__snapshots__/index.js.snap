// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AndroidBack should match snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
